SpatialGDK/Source/SpatialGDKEditor/Private/SpatialTestSettings.cpp

```diff
@@ -0,0 +1,103 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+#include "SpatialTestSettings.h"
+
+FSpatialTestSettings::FSpatialTestSettings()
+	: OriginalSpatialGDKSettings(nullptr)
+	, OriginalLevelEditorPlaySettings(nullptr)
+	, OriginalSpatialGDKEditorSettings(nullptr)
+	, OriginalGeneralProjectSettings(nullptr)
+	, OriginalEditorPerformanceSettings(nullptr)
+{
+}
+
+void FSpatialTestSettings::Override(const FString& MapName)
+{
+	// Back up the existing settings so they can be reverted later
+	Backup();
+	// First override the settings from the base config file, if it exists
+	Load(BaseOverridesFilename);
+	// Then override the settings from the map specific config file, if it exists
+	FString MapOverridesFilename = OverrideSettingsBaseFilename + FPackageName::GetShortName(MapName) + (OverrideSettingsFileExtension);
+	Load(MapOverridesFilename);
+}
+
+void FSpatialTestSettings::Backup()
+{
+	// Duplicate original settings but use different outer - if same outer is reused the object is not duplicated and pointer is to the same
+	// Object Having the same name causes runtime exceptions
+	// Use additional object flag RF_Standalone to avoid early GC - destroy when restoring settings
+	OriginalSpatialGDKSettings = NewObject<USpatialGDKSettings>(
```

Comment:
Nit: I guess this could be refactored slightly with a local templated function, similar to what I have added for `UGeneratedTestMap::AddActorToLevel`, just so the USpatialGDKSettings and `EObjectFlags(RF_Public | RF_ClassDefaultObject | RF_ArchetypeObject | RF_Standalone)`, etc. is not repeated.

Reply:
Yes, that makes the duplication much more compact and I've done the same for the restore too.